File: 700. Search in a Binary Search Tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func searchBST(root *TreeNode, val int) *TreeNode {

	if root == nil || root.Val == val {
		return root
	}

	var node *TreeNode = searchBST(root.Left, val)

	if node == nil {
		node = searchBST(root.Right, val)
	}

	return node
}

func main() {
}

File: 700. Search in a Binary Search Tree.py
class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if root == None or root.val == val: return root
        left = self.searchBST(root.left,val)
        return left if left != None else self.searchBST(root.right,val)